### PR TITLE
uprava ukazkoveho prikladu pouziti dle zadani

### PR DESCRIPTION
**Pull request by bageta**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -6,51 +6,46 @@ using std::cout;
...
+                "Enable verbose logging");
+
+  args::AddInt(&size, {'s', "size"}, args::OPTIONAL,
+               "Size of something");
 
   // Parse command line flags. Die on error.
   args::Parse(&argc, &argv);
-
+  
+  if (verbose.present()){
```
Comment (agentydragon) [inline main.cpp]: Please add space before {.

File: main.cpp [hunk 2]
```diff
@@ -6,51 +6,46 @@ using std::cout;
...
+    printf("verbose = true");
+  } else {
+    printf("verbose = false");
```
Comment (agentydragon) [inline main.cpp]: Please use std::cout.

File: main.cpp (hunk 1 above)
Comment (agentydragon) [inline main.cpp]: Also, the condition should read "verbose.present() && verbose.get()" to guard against --verbose=false.

File: main.cpp (hunk 2 above)
Comment (agentydragon) [inline main.cpp]: And newlines.

File: main.cpp [hunk 3]
```diff
@@ -6,51 +6,46 @@ using std::cout;
...
   // (usually the program name) and nonflags arguments.
-  std::vector<char*> synchronized_paths(argc);
-  copy(&argv[1], &argv[argc], synchronized_paths.begin());  // skip [0]
-  SetSynchronizedFiles(synchronized_paths);
-
-  ConnectWithTimeout(hostname.get(), protocol.get(),
-                     timeout_ms.present() ? timeout_ms.get() : 1000);
-  // ...
+  std::vector<char*> non_flags(argc);
+  copy(&argv[1], &argv[argc], non_flags.begin()); //skip 0
```
Comment (agentydragon) [inline main.cpp]: Please add space before 'skip'. Ditto below.

File: main.cpp [hunk 4]
```diff
@@ -6,51 +6,46 @@ using std::cout;
...
+
+  //print non flags arguments
+  cout << "args =";
+  for (auto non_flag : non_flags){
```
Comment (agentydragon) [inline main.cpp]: const auto&

**Comment by bageta** [comment]
done

File: main.cpp [hunk 5]
```diff
@@ -6,51 +6,47 @@ using std::cout;
...
-                  "Hostname of the synchronization server.");
-  args::AddEnum(&protocol, {"protocol", 'p'}, args::REQUIRED,
-                "Synchronization protocol to use.",
-                {"http", "ftp", "https"});
-  args::AddInt(&timeout_ms, 't', args::OPTIONAL,
-               "Connection timeout in ms. Default value is 1000.",
-               0, 60000);
-  args::AddBool(&verbose, {"verbose", 'v'}, args::OPTIONAL,
-                "Enable verbose logging?");
+  args::AddBool(&verbose, {'v', "verbose"}, args::FlagUse::OPTIONAL,
```
Comment (agentydragon) [inline main.cpp]: Does this actually compile? (I would expect 'args::OPTIONAL' to work here...)

**Comment by agentydragon** [comment]
LGTM otherwise

**Comment by agentydragon** [comment]
Looks good. I still think `args::REQUIRED` is slightly better than `args::FlagUse::REQUIRED`.

**Pull request merged**